data_model/src/domain.rs

```diff
@@ -71,16 +72,17 @@ impl From<GenesisDomain> for Domain {
 
 /// Builder which can be submitted in a transaction to create a new [`Domain`]
 #[derive(
-    Debug, Display, Clone, PartialEq, Eq, Decode, Encode, Deserialize, Serialize, IntoSchema,
+    Debug, Display, Clone, IdOrdEqHash, Decode, Encode, Deserialize, Serialize, IntoSchema,
 )]
 #[allow(clippy::multiple_inherent_impl)]
 #[display(fmt = "[{id}]")]
+#[id(type = "<Domain as Identifiable>::Id")]
 pub struct NewDomain {
-    /// The identification associated to the domain builder.
+    /// The identification associated with the domain builder.
     id: <Domain as Identifiable>::Id,
     /// The (IPFS) link to the logo of this domain.
     logo: Option<IpfsPath>,
-    /// metadata associated to the domain builder.
+    /// Metadata associated with the domain builder.
     metadata: Metadata,
 }
```

Comment:
Why can't we infer type from `id` field? Can `#[id(type = "...")]` be optional?

Reply:
Created an [issue](https://github.com/hyperledger/iroha/issues/2521).